dspace-api/src/main/java/org/dspace/content/SiteServiceImpl.java

```diff
@@ -50,4 +58,33 @@ public Site createSite(Context context) throws SQLException {
     public Site findSite(Context context) throws SQLException {
         return siteDAO.findSite(context);
     }
+
+    @Override
+    public Site find(Context context, UUID id) throws SQLException {
+        return siteDAO.findByID(context, Site.class, id);
+    }
+
+    @Override
+    public void updateLastModified(Context context, Site dso) throws SQLException, AuthorizeException {
+        //Not used at the moment
+    }
+
+    @Override
+    public void update(Context context, Site dso) throws SQLException, AuthorizeException {
+        if(!authorizeService.isAdmin(context)){
+            throw new AuthorizeException();
+        }
+        siteDAO.save(context, dso);
```

Comment:
We should trigger an event here. IIRC at least the RDFConsumer would consume the event.